src/FeeRouter/RewardRouter.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+interface IInbox {
+    function createRetryableTicket(
+        address to,
+        uint256 l2CallValue,
+        uint256 maxSubmissionCost,
+        address excessFeeRefundAddress,
+        address callValueRefundAddress,
+        uint256 gasLimit,
+        uint256 maxFeePerGas,
+        bytes calldata data
+    ) external payable returns (uint256);
+}
+
+contract RewardRouter {
+    IInbox immutable inbox;
+    address immutable destination;
+
+    constructor(address _inbox, address _destination) {
+        inbox = IInbox(_inbox);
+        destination = _destination;
+    }
+
+    receive() external payable {}
+
+    function routeFunds(
+        uint256 maxSubmissionCost,
+        uint256 gasLimit,
+        uint256 maxFeePerGas
+    ) external payable {
+        _routeFunds(maxSubmissionCost, gasLimit, maxFeePerGas, msg.sender);
+    }
+
+    function routeFundsCustomRefund(
+        uint256 maxSubmissionCost,
+        uint256 gasLimit,
+        uint256 maxFeePerGas,
+        address excessFeeRefundAddress
+    ) external payable {
+        _routeFunds(
+            maxSubmissionCost,
+            gasLimit,
+            maxFeePerGas,
+            excessFeeRefundAddress
+        );
+    }
+
+    function _routeFunds(
+        uint256 maxSubmissionCost,
+        uint256 gasLimit,
+        uint256 maxFeePerGas,
+        address excessFeeRefundAddress
+    ) internal {
+        require(
```

Comment:
I think we should include a comment as to why this is necessary (to stop someone spending all the value on gas), given that we also do similar kind of check in the inbox